scripts/check-go-version.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+version_regex="[0-9]\.[0-9]"
+update=false
+
+while getopts "u" opt; do
+    case $opt in
+        u) update=true ;;
+        *) echo "Usage: $0 [-u]" >&2
+           exit 1
+           ;;
+    esac
+done
+
+# Fetch latest go release version
+go_latest_version=$(curl -s https://go.dev/dl/?mode=json | jq -r '.[0].version' | awk -F'.' '{gsub("go", ""); print $1"."$2}')
```

Comment:
I suggest parsing it here into `major` and `minor` to simplify the rest of the code